### PR TITLE
Add documentation of the billing system

### PR DESCRIPTION
**Pull request by schmie**
What
----

As part of new billing we have generated some information on how it works.

How to review
-------------

Have a look at the changes and whether you think the information is in the right place - might be one for product pages?

1. Preview the content locally ([see README](https://github.com/alphagov/paas-tech-docs#preview)) and check that it renders as expected.
1. Manually check that any removed or renamed anchor tags are not in use ([linkchecker](https://github.com/linkcheck/linkchecker) doesn't do this).
1. …

Who can review
--------------

Describe who can review the changes. Or more importantly, list the people
that can't review, because they worked on it.

### CODE REVIEW

File: source/billing_system.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# The GOV.UK PaaS billing system
+
+The GOV.UK Platform as a Service (PaaS) billing system generates bills for GOV.UK PaaS tenants. Our tenants use GOV.UK PaaS to create and delete various services and resources, such as compute instances running applications and databases. We need to calculate how much these services and resources cost on a monthly basis so we can recoup this money from tenants.
+
+The GOV.UK PaaS billing system comprises:
+
+- 2 applications written in Golang: the `paas-billing-collector` and the `paas-billing-api`
```
Comment (whpearson) [inline source/billing_system.html.md.erb:7]: Might be worth saying that both applications use the same codebase?

File: source/billing_system.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+- the GOV.UK PaaS billing Postgres database application, which uses stored functions to calculate tenants' bills in the database
+
+## Accessing GOV.UK PaaS billing
+
+To access GOV.UK PaaS billing information, use the PaaS Admin web application through the *Explore your costs and usage* link on your organisation’s page. You can also access more detailed information, such as resource IDs and charging plan details, through the [PaaS Billing API](https://github.com/alphagov/paas-billing/README.md).
+
+PaaS Admin calls the `\billable_events` endpoint of the API. The API then queries the billing database to get the required data and calls a stored function. For example, [the `get_tenant_bill` stored function](https://github.com/alphagov/paas-billing/blob/main/billing-db/sprocs/get_tenant_bill.sql)) is the database entry point for calculating tenant bills.
+
+
```
Comment (whpearson) [inline source/billing_system.html.md.erb:16]: Mention something about admin level stuff found in https://admin.london.cloud.service.gov.uk/platform-admin ?

File: source/billing_system.html.md.erb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    This stored function takes an organisation name as well as the start and end date of the queried time frame as arguments. It then goes through the *resources* table for the tenant and time frame, collects records of which services or resources the tenants used, and writes them to a temporary table.
+
+2. Calculating the bill
+
+    The [`calculate_bill`](https://github.com/alphagov/paas-billing/blob/main/billing-db/sprocs/calculate_bill.sql) stored function calculates the bill. This function accepts no parameters. It takes the contents of the temporary table populated by `get_tenant_bill` and calculates the bill, taking account of factors such as VAT rates and currency exchange rates. The exchange rates are important because many services provisioned by GOV.UK PaaS are billed in US dollars (USD), but we produce bills for our tenants in British pounds (GBP).
```
Comment (whpearson) [inline source/billing_system.html.md.erb:29]: Isn't this the other way round? get_tenant_bill calls calculate_bill

File: source/billing_system.html.md.erb [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+### How we calculate bills
+
+The PaaS billing system calculates tenants’ costs, based on pricing information collected from our service providers. We do this by writing simple formulae in the billing database, and each service or resource we provide to tenants has a formula to calculate the cost.
+
+
+```
+($storage_in_mb/1024) * ($time_in_seconds/2678401) * 1.23
+```
+
```
Comment (whpearson) [inline source/billing_system.html.md.erb:40]: Might be worth picking an example that uses $external_price ?

**Review by whpearson — CHANGES_REQUESTED** [review]
A few comments. It would be good to get someone without much billing experience to give this the once over too.

File: source/billing_system.html.md.erb (hunk 1 above)
Comment (schmie) [reply source/billing_system.html.md.erb]: Good shout! Added in a comment and link to the paas-billing repo

File: source/billing_system.html.md.erb (hunk 4 above)
Comment (schmie) [reply source/billing_system.html.md.erb:40]: Added that in, as well as a comment that we resolve usage to the second.

File: source/billing_system.html.md.erb (hunk 3 above)
Comment (schmie) [reply source/billing_system.html.md.erb]: I reckon this focuses more on the data flow in that it "takes the temp table" - I'm unsure if adding the call structure here does not complicate things without adding much?

**Review by whpearson — APPROVED** [review]
Looks good

File: source/billing_system.html.md.erb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    The [`calculate_bill`](https://github.com/alphagov/paas-billing/blob/main/billing-db/sprocs/calculate_bill.sql) stored function calculates the bill. This function accepts no parameters. It takes the contents of the temporary table populated by `get_tenant_bill_api` and calculates the bill, taking account of factors such as VAT rates and currency exchange rates. The exchange rates are important because many services provisioned by GOV.UK PaaS are billed in US dollars (USD), but we produce bills for our tenants in British pounds (GBP).
+
+
+### How we calculate bills
+
+The PaaS billing system calculates tenants’ costs, based on pricing information collected from our service providers. We do this by writing simple formulae in the billing database, and each service or resource we provide to tenants has a formula to calculate the cost.
+
+
+```
+($storage_in_mb/1024) * ($time_in_seconds/2678401) * $external_price
```
Comment (cadmiumcat) [inline source/billing_system.html.md.erb:38]: If it's a different formula for every resource, is it worth saying that this is an example?
If it's the same, then maybe clarify that this is it

Also, I would finder clearer if this was an equation, e.g.:
```suggestion
Total charge = ($storage_in_mb/1024) * ($time_in_seconds/2678401) * $external_price
```

**Review by cadmiumcat — APPROVED** [review]

**Review by paroxp — APPROVED** [review]

**Pull request merged**